### ISSUE TITLE
substringToIndex crash

### ISSUE THREAD

**Issue by rafalkitta** [sensorberg-dev/ios-sdk#19]
### Classification (type of problem: security, crash, performance, serious bug, feature, enhancement etc)
Crash - about 30% of all users!

### Reproducibility (always, sometimes, rarely, N.A.)
N.A
Didn't try to reproduce crash under debugger.

### Area (scanning, network etc)
Initialising SBMGeofence.

### Description
Crash description from Crashlytics:
Fatal Exception: NSRangeException
*** -[__NSCFConstantString substringToIndex:]: Index 8 out of bounds; string length 0

Fatal Exception: NSRangeException
0  CoreFoundation                 0x18ff051b8 __exceptionPreprocess
1  libobjc.A.dylib                0x18e93c55c objc_exception_throw
2  CoreFoundation                 0x18ff05100 -[NSException initWithCoder:]
3  Foundation                     0x19092ae80 -[NSString substringToIndex:]
4  SensorbergSDK                  0x1016c0768 (Missing)
5  SensorbergSDK                  0x1016bb174 (Missing)
6  CoreLocation                   0x197f095f4 (null)
7  CoreLocation                   0x197efb4b8 (null)
8  CoreFoundation                 0x18feb2a44 __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__
9  CoreFoundation                 0x18feb2240 __CFRunLoopDoBlocks
10 CoreFoundation                 0x18feb0094 __CFRunLoopRun
11 CoreFoundation                 0x18fdde2b8 CFRunLoopRunSpecific
12 GraphicsServices               0x191892198 GSEventRunModal
13 UIKit                          0x195e257fc -[UIApplication _run]
14 UIKit                          0x195e20534 UIApplicationMain
15 AppName                          0x100217f84 main (AppDelegate.swift:24)
16 libdispatch.dylib              0x18edc15b8 (Missing)

 Seems like `geohash` init param contains empty string.
Line 121 in SBModel.m

### Sensorberg SDK version
2.5.1

### iOS version
8, 9, 10

### Device
iPhone, iPad

### Additional notes
Probably infected line of code:
```objc
GHArea *area = [GeoHash areaForHash:[geohash substringToIndex:8]];
```
When crash occurs `geohash` variable is empty.

**Comment by tagyro** [comment]
Hi Rafał
Thanks for reporting this, we'll investigate it!

**Comment by tagyro** [comment]
Hi Rafał
This issue has been fixed with #0017f7b77b16cd4494de9a88720cc404eafa6f66 and fixed in [v2.5.3](https://github.com/sensorberg-dev/ios-sdk/releases/tag/2.5.3)
